docs/proposals/performance.md

```diff
@@ -0,0 +1,21 @@
+# Post-1.0 performance/scalability improvement ideas
+
+This document is a collection of ideas for improving Kubernetes
+performance/scalability post-1.0. Feel free to add to it. Please
+link to relevant issues and assign an owner. Please only add things
+that will actually improve performance.
+
+## Performance/scalability goal
+
+The next performance/scalability goal is 1000 nodes with 30 pods/node. We may wish to revisit the 30 pods/node part if we find that this is far above what real users need.
+
+## Ideas
+
+1. Change communication between apiserver and etcd to proto-based once v3 etcd is available [#8132](https://github.com/GoogleCloudPlatform/kubernetes/issues/8132) @wojtek-t
+2. Use BSON (http://bsonspec.org/) [#8132](https://github.com/GoogleCloudPlatform/kubernetes/issues/8132) @lavalamp
+3. Eliminate "internal" representation and thus conversions [#8190](https://github.com/GoogleCloudPlatform/kubernetes/issues/8132) @bgrant0607
+4. Alternative JSON parsers [#3338](https://github.com/GoogleCloudPlatform/kubernetes/issues/3338) @lavalamp
+5. Watch improvements
+   - implement watch at API server level [#8132](https://github.com/GoogleCloudPlatform/kubernetes/issues/8132) @wojtek-t
```

Comment:
Yeah, since the meta issue isn't related to protocol buffers. 

IMO the basic building blocks we'd need to fix watch post 1.0 are:
1. Single watch on / from apiserver to etcd 
2. A store that is resource version/change history aware
3. An index on predicate functions 

Whatever else the etcd v3 api gives us (larger watch window, streaming changes etc) is only going to complement this.  The comment you mentioned captures a lot of this  https://github.com/GoogleCloudPlatform/kubernetes/issues/4817#issuecomment-89437368 so it's probably better to use that.